docs/src/tutorials/device-monitoring.md

```diff
@@ -0,0 +1,154 @@
+
+Thin-edge.io uses the open source component [`collectd`](https://collectd.org/) to collect the metrics from the device.
+Thin-edge.io translates the collected metrics from their native format to the [thin-edge.io JSON](../architecture/thin-edge-json.md) format
+and then into the [cloud-vendor specific format](../architecture/mapper.md).
+
+Enabling monitoring on your device is a 3-steps process:
+1. [Install `collectd`](#install-collectd),
+1. [Configure `collectd`](#configure-collectd),
+1. [Enable thin-edge.io monitoring](#enable-thin-edge-monitoring).
+
+## Install `collectd`
+
+Device monitoring is not enabled by default when you install thin edge.
+You will have to install and configure [`collectd`](https://collectd.org/) first.
+
+To install collectd, follow the [collectd installation process](https://collectd.org/download.shtml)
+that is specific to your device. On a Debian or Ubuntu linux:
+
+``` shell
+sudo apt-get install collectd-core
+```
+
+## Configure `collectd`
+
+### TLDR; Just want it running
+
+Thin-edge.io provides a [basic `collectd` configuration](https://github.com/thin-edge/thin-edge.io/blob/main/configuration/contrib/collectd/tedge-collectd.conf)
+that can be used to collect cpu, memory and disk metrics.
+
+Simply copy that file to the main collectd configuration file and restart the daemon
+(it might be good to keep a copy of the original configuration).
+
+``` shell
+sudo cp /etc/tedge/contrib/collectd/collectd.conf sudo cp /etc/tedge/contrib/collectd/collectd.conf.backup
+sudo cp /etc/tedge/contrib/collectd/collectd.conf /etc/collectd/collectd.conf
+sudo systemctl restart collectd
+```
+
+### `Collectd.conf`
+
+The behavior of collectd is defined by the 
+[`collectd.conf` configuration file](https://collectd.org/documentation/manpages/collectd.conf.5.shtml)
+(usually located at `/etc/collectd/collectd.conf`)
+
+__Important notes__ You can enable or disable the collectd plugins of your choice, but with some notable exceptions:
+1. __MQTT must be enabled__.
+   * Thin-edge.io expects the collectd metrics to be published on the local MQTT bus.
+     Hence, you must enable the [MQTT write plugin of collectd](https://collectd.org/documentation/manpages/collectd.conf.5.shtml#plugin_mqtt).
+   * The MQTT plugin is available on most distribution of `collectd`, but this is the case on MacOS using homebrew.
+     If you are missing the MQTT plugin, please recompile `collectd` to include the MQTT plugin.
+     See [https://github.com/collectd/collectd](https://github.com/collectd/collectd) for details.
+   * Here is a config snippet to configure the MQTT write plugin:
+     ```
+        LoadPlugin mqtt
+        
+        <Plugin mqtt>
+            <Publish "tedge">
+                Host "localhost"
+                Port 1883
+                ClientId "tedge-dm"
+            </Publish>
+        </Plugin>
+     ```
```

Comment:
We might want to add one extra line that this configuration snippet is already included in the basic collectd configuration that we provide and hence he doesn't need to add it. This snippet needs to be added only if the user has his own collectd.conf file.

Reply:
I added an intro "Unless you opted for the [minimal test configuration provided with thin-edge"